armi/materials/material.py

```diff
@@ -151,6 +151,12 @@ def getChildren(
         """Return empty list, representing that materials have no children."""
         return []
 
+    def getGrandparent(self):
+        """Return the grandparent (typically block) associated with this material"""
+        # would be nice to use composite.getAncestor, but material is not a composite
+        component = self.parent
+        return None if component is None else component.parent
```

Comment:
I think you can replace this with `component.getAncestor(lambda x: isinstance(x, Block))`